src/main/java/subway/service/LineService.java

```diff
@@ -0,0 +1,69 @@
+package subway.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.domain.Line;
+import subway.domain.Station;
+import subway.dto.LineRequest;
+import subway.dto.LineResponse;
+import subway.repository.LineRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+    LineRepository lineRepository;
+    StationService stationService;
+
+    public LineService(LineRepository lineRepository, StationService stationService) {
+        this.lineRepository = lineRepository;
+        this.stationService = stationService;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Line line = lineRepository.save(new Line(lineRequest.getName(), lineRequest.getColor(), lineRequest.getUpStationId(), lineRequest.getDownStationId(), lineRequest.getDistance()));
+        return createLineResponse(line);
+    }
+
+    public List<LineResponse> findAllLines(){
+        return lineRepository.findAll().stream()
+                .map(this::createLineResponse)
+                .collect(Collectors.toList());
+    }
+
+    private LineResponse createLineResponse(Line line) {
+        List<Station> stationList = new ArrayList<>();
+        stationList.add(stationService.findById(line.getUpStationId()));
+        stationList.add(stationService.findById(line.getDownStationId()));
+
+        return new LineResponse(line.getId(), line.getName(), line.getColor(), stationList);
+    }
+
+    public LineResponse findLineById(Long id) {
+        return lineRepository.findById(id).map(this::createLineResponse).get();
+    }
+
+    @Transactional
+    public void deleteLineById(Long id) {
+        lineRepository.deleteById(id);
+    }
+
+    @Transactional
+    public void updateLine(Long id, LineRequest lineRequest) {
+        Line line = lineRepository.findById(id).get();
+
+        if (! line.getName().equals(lineRequest.getName())) {
+            line.changeName(lineRequest.getName());
+        }
+
+        if (! line.getColor().equals(lineRequest.getColor())) {
+            line.changeColor(lineRequest.getColor());
+        }
```

Comment:
Line 업데이트 시 name과 color를 동일한 시점에 변경해주고있는데요. 
name과 color가 하나의 메서드에서 함께 수정되어도 될 것 같은데 어떻게 생각하시나요~?🤔 

그렇게 된다면 Line을 업데이트시 `name과 color 가 함께 수정되어야 한다 라는 비즈니스`를 나타낼 수 있을 것 같습니다 :-) 

```suggestion
            line.change(lineRequest.getName(), lineRequest.getColor()); 
```